gems/smithy/lib/smithy/views/client/types.rb

```diff
@@ -88,9 +83,42 @@ def attribute_docstrings
             docstrings.each do |docstring|
               lines << "  #{docstring}"
             end
-            lines << "  @return [#{Model::YARD.type(@model, @id, @target)}]"
+            lines << "  @return [#{@doc_type}]"
             lines
           end
+
+          def default?
+            traits = @member.fetch('traits', {})
+            traits.key?('smithy.api#default') && !traits.key?('smithy.api#clientOptional')
```

Comment:
This gets a little confusing for shape only gems / server generation - the smithy docs say 
> The [clientOptional trait](https://smithy.io/2.0/spec/type-refinement-traits.html#clientoptional-trait) applied to a member marked with the default trait causes [non-authoritative generators](https://smithy.io/2.0/guides/building-codegen/overview-and-concepts.html#client-generation) to ignore the default trait.

So, in a client context, we should use it, but in an authoritative (server) type we shouldn't.  I'm not sure if theres a good way to handle that in a server/client agnostic generation though and since we're targeting client generation, maybe this is good for now.

Reply:
Yeah. Good call out.. I'm not sure how else to handle this? I'll chat with Michael.